File: Python3/96.unique-binary-search-trees.py
#
# @lc app=leetcode id=96 lang=python3
#
# [96] Unique Binary Search Trees
#

# @lc code=start
class Solution:
    def numTrees(self, n: int):
        if n < 2:
            return n
        dp = [0] * n
        

        
# @lc code=end

File: C#/96.unique-binary-search-trees.cs
/*
 * @lc app=leetcode id=96 lang=csharp
 *
 * [96] Unique Binary Search Trees
 */
using System.Collections.Generic;
// @lc code=start
public class Solution {
    public int NumTrees(int n) {
        List<int> dp = new List<int>();
        dp.Add(1);
        dp.Add(1);
        for(int curend = 1;curend < n; curend++){
            int Cnum = 0;
            for(int i = 0; i <= curend; i++){
                Cnum += dp[i] * dp[curend - i];
            }
            dp.Add(Cnum);
        }
        return dp[n];
    }
}
// @lc code=end

